pkg/csi/service/vanilla/controller.go

```diff
@@ -269,20 +269,24 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
 			for _, authMgr := range authMgrs {
 				go common.ComputeDatastoreMapForBlockVolumes(authMgr, config.Global.CSIAuthCheckIntervalInMin)
 			}
-			if !multivCenterTopologyDeployment {
-				for _, vcconfig := range c.managers.VcenterConfigs {
-					isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
-						vcconfig.Host)
-					if err != nil {
-						return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
-							"Error:%+v", vcconfig.Host, err)
-					}
-					if isvSANFileServicesSupported {
-						for _, authMgr := range authMgrs {
-							go common.ComputeFSEnabledClustersToDsMap(authMgr, config.Global.CSIAuthCheckIntervalInMin)
-						}
-					}
+			var vsanFileServiceNotSupported bool
+			for _, vcconfig := range c.managers.VcenterConfigs {
+				isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
+					vcconfig.Host)
+				if err != nil {
+					return logger.LogNewErrorf(log, "failed to verify if vSAN file services is supported or not for vCenter: %s. "+
+						"Error:%+v", vcconfig.Host, err)
 				}
+				if !isvSANFileServicesSupported {
+					vsanFileServiceNotSupported = true
+					break
+				}
+			}
+			if vsanFileServiceNotSupported {
+				return logger.LogNewErrorf(log, "vSAN file service is not supported in one or more vCenter(s)")
```

Comment:
I think we asked FVT to verify the case where FS is not enabled on one of the VCs. Is that not a valid case?

Reply:
@divyenpatel Can you comment on this ?

Reply:
This is not a FS enabled check, it is just a VC version check